src/app/Services/shop-list.service.ts

```diff
@@ -1,31 +1,38 @@
 import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ShopListService {
-  ingredients = [
+  ingredients = new BehaviorSubject<Array<any>>([
     { id: 1, name: 'Bread', amount: 4 },
     { id: 2, name: 'Tomato', amount: 6 },
     { id: 3, name: 'Vegetable', amount: 8 },
-  ];
+  ]);
 
   constructor() {}
 
   addIngredient(ingredient: any) {
-    this.ingredients.push(ingredient);
+    const newIngredients = [...this.ingredients.value, ingredient];
+    this.ingredients.next(newIngredients);
   }
 
   updateIngredient(ingredient: any) {
-    this.ingredients.forEach((item) => {
-      if (item.id === ingredient.id) {
-        item.name = ingredient.name;
-        item.amount = ingredient.amount;
+    const length = this.ingredients.value.length;
+    for (let i = 0; i < length; i++) {
+      if (this.ingredients.value[i].id === ingredient.id) {
```

Comment:
prefer using `for...of` (haha) cuz we using the length of that array, not a special length.
and create `const ingredient = this.ingredients.value[i]`

Reply:
![image](https://user-images.githubusercontent.com/47804324/168637581-4bc665db-3e7d-411e-a75a-52335d166855.png)